From e80aad8053981a7305bf89bb49f7a8a829bb8656 Mon Sep 17 00:00:00 2001
From: Liang Li <Liang.Li@windriver.com>
Date: Wed, 3 Dec 2008 17:49:09 +0800
Subject: [PATCH] Fix watchdog driver bug of mpc5200_wdt

Using the watchdog as character device under /dev.The write method of
the watchdog driver should not return 0 in all conditions.We can simply
reproduce the bug while the write method of a char device driver return
0 with following command:

[root@localhost]#echo 0 > /dev/watchdog

As dummy write method of watchdog device,simply return length of requested
data would be sufficient.

Signed-off-by: Liang Li <Liang.Li@windriver.com>
---
 drivers/watchdog/mpc5200_wdt.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/watchdog/mpc5200_wdt.c b/drivers/watchdog/mpc5200_wdt.c
index db91892..80ad031 100644
--- a/drivers/watchdog/mpc5200_wdt.c
+++ b/drivers/watchdog/mpc5200_wdt.c
@@ -88,7 +88,7 @@ static ssize_t mpc5200_wdt_write(struct file *file, const char __user *data,
 {
 	struct mpc5200_wdt *wdt = file->private_data;
 	mpc5200_wdt_ping(wdt);
-	return 0;
+	return len;
 }
 static struct watchdog_info mpc5200_wdt_info = {
 	.options	= WDIOF_SETTIMEOUT | WDIOF_KEEPALIVEPING,
-- 
1.6.0.2.GIT

